### ISSUE TITLE
bad "where" clause indentation

### ISSUE THREAD

**Issue by ensc** [rust-lang/rust-mode#257]
Styleguide says that the `where` keyword must be at the same level like `fn`, e.g.

```rust
fn foo<T>()
where
    T: Bar,
{
}
```

But rust-mode 20171208 formats this as

```rust
fn foo<T>()
    where
    T: Bar,
{
}

```

**Comment by brotzeit** [comment]
The formatting is done by rustfmt.

**Comment by ensc** [comment]
really?  rustfmt is not installed here but indentation by `<TAB>` or `indent-region` works.

**Comment by brotzeit** [comment]
I misunderstood you. Unfortunately I don't know how rust indentation works, but I wouldn't care too much about it and start using rustfmt as it takes care of these things for you.

**Comment by sebastiencs** [comment]
@ensc You can set rust-indent-where-clause to nil:
```el
(setq rust-indent-where-clause nil)
```

**Comment by ensc** [comment]
@sebastiencs thx, works

@brotzeit internal formatting (`<TAB>`, `indent-region`) should not alter a `rustfmt` formatted code (at least for trivial and common cases).  Else, you will always write code in the "wrong" style, need to reformat it explicitly and on next `<TAB>` it will be "wrong" again.

**Comment by brotzeit** [comment]
That's the reason why I don't format/indent rust code manually. I think I don't use tab at all.

**Comment by tromey** [comment]
> You need to set rust-indent-where-clause to nil

If this is the standard then I think it should be the default.

**Comment by nikomatsakis** [comment]
I agree we should change the default.

**Comment by kelseasy** [comment]
As this was solved in #258, I think this can be closed!

**Comment by tromey** [comment]
Thanks for the ping.
